spec/controllers/troubleshooting_controller_spec.rb

```diff
@@ -0,0 +1,253 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe TroubleshootingController do
+  include LoginMacros
+  include RedirectExpectationHelper
+
+  let(:tag) { create(:canonical_fandom) }
+  let(:work) { create(:posted_work) }
+  let(:tag_wrangler) { create(:user, roles: [Role.new(name: "tag_wrangler")]) }
+  let(:non_tag_wrangler) { create(:user) }
+
+  context "show" do
+    context "when logged out" do
+      it "errors and redirects to the login page" do
+        get :show, params: { tag_id: tag.to_param }
+        it_redirects_to_with_error(new_user_session_path, "Sorry, you don't have permission to access the page you were trying to reach. Please log in.")
+      end
+    end
+
+    context "when logged in as a regular user" do
+      before { fake_login_known_user(non_tag_wrangler) }
+
+      it "errors and redirects to user's profile" do
+        get :show, params: { tag_id: tag.to_param }
+        it_redirects_to_with_error(non_tag_wrangler, "Sorry, you don't have permission to access the page you were trying to reach.")
+      end
+    end
+
+    context "when logged in as a tag wrangler" do
+      before { fake_login_known_user(tag_wrangler) }
+
+      it "shows a form with the allowed actions for tags" do
+        get :show, params: { tag_id: tag.to_param }
+        expect(response).to render_template(:show)
+        expect(assigns[:allowed_actions]).to \
+          contain_exactly(*%w[fix_associations fix_counts fix_meta_tags])
```

Comment:
I think you can get around this warning by using `eq` the array.

Reply:
Hmm, I wanted to use `contain_exactly` so that the tests won't fail if someone decides to reorder the allowed actions (since the order affects how they're displayed). I guess I'll avoid `%w` to try to make Hound happy here.